iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/variant_internal.hpp

```diff
@@ -245,6 +256,20 @@ struct call_at_index<N, T>
             assert(false && "Could not call copy constructor for variant element");
         }
     }
+
+    // NOLINTJUSTIFICATION 'operator new()' needs non-const 'destination'
+    // NOLINTNEXTLINE(readability-non-const-parameter)
+    static bool equality(const uint64_t index, const byte_t* lhs, const byte_t* rhs) noexcept
+    {
+        if (N == index)
+        {
+            // AXIVION Next Construct AutosarC++19_03-A5.2.4 : Type safety ensured through template parameter and encapsulated in this class
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+            return *reinterpret_cast<const T*>(lhs) == *reinterpret_cast<const T*>(rhs);
+        }
+        assert(false && "Could not call equality operator for variant element");
```

Comment:
Here I would call `Expects`. This is a hard contract, if we reach this point here we screwed up our implementation since the variant should ensure beforehand that the type is contained in the variant.